SpringSchool/src/main/java/com/example/springschool/controllers/SchoolController.java

```diff
@@ -0,0 +1,37 @@
+package com.example.springschool.controllers;
+
+import com.example.springschool.entities.School;
+import com.example.springschool.entities.Student;
+import com.example.springschool.services.SchoolService;
+import lombok.AllArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@CrossOrigin(origins = "*")
+@RequestMapping(value = "/schools")
+@RestController
+@AllArgsConstructor
+public class SchoolController {
+    SchoolService schoolService;
+
+    @GetMapping()
+    public List<School> getAllSchools() {
+        return this.schoolService.getAllSchools();
+    }
+
+    @GetMapping(value = "/{id}")
+    public School getSchoolById(@PathVariable Long id) {
+        return this.schoolService.getSchoolByIdOrThrow(id);
+    }
+
+    @GetMapping(value = "{schoolId}/students")
+    public List<Student> getAllStudentsById(@PathVariable Long schoolId) {
+        return this.schoolService.getAllStudentsById(schoolId);
+    }
+
+    @GetMapping(value = "{schoolId}/students/{studentId}")
+    public Student getStudentBySchoolIdByStudentId(@PathVariable Long schoolId, @PathVariable Long studentId) {
```

Comment:
```suggestion
    public Student getStudentBySchoolIdAndStudentId(@PathVariable Long schoolId, @PathVariable Long studentId) {
```

du kartus By dažniausiai nesirašo, po By tiesiog vardinamos sąlygos :)

Reply:
tiesa